src/scanner/exposed-apis.ts

```diff
@@ -36,15 +36,14 @@ export let scan = (
         documentUtils,
         messageDecorator,
         (ruleId, axeHelpUrl) => helpUrlGetter.getHelpUrl(ruleId, axeHelpUrl),
+        ruleToLinkConfiguration,
     );
     const launcher = new Launcher(axe, scanParameterGenerator, document, options);
     const axeResponseHandler = new AxeResponseHandler(
         successCallback,
         errorCallback,
```

Comment:
With the `setRuleToLinksConfiguration()` call being deleted below, what other purpose does `resultDecorator` serve?

Reply:
having a separate method setRuleToLinksConfiguration to set a dependency that is mandatory for resultDecorator to work is error prone. Instead with this pr, i am passing the ruleToLinkConfiguration in constructor. Also, this object will never change anyway.

Reply:
see integration.test.ts.snap. this will tell you what i am fixing.

Reply:
Oh I see what you're fixing, but I was wondering what else `resultDecorator` did in this file and if this instance of it could be deleted